sjb/config/test_cases/ci-kubernetes-machine-api-operator.yml

```diff
@@ -0,0 +1,232 @@
+---
+parent: 'common/test_cases/minimal.yml'
+extensions:
+  actions:
+    - type: "host_script"
+      title: "Upload the default AWS credentiasl"
+      script: |-
+         ssh -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config openshiftdevel "mkdir ~/.aws"
+         scp -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config ~/.aws/credentials openshiftdevel:~/.aws
+    - type: "script"
+      title: "Install minikube"
+      script: |-
+        # Install minikube
+        curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.28.0/minikube-linux-amd64
+        chmod +x minikube
+        sudo mv minikube /usr/bin/
+        # Install kubectl
+        curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl
+        chmod +x kubectl
+        sudo mv kubectl /usr/bin/
+    - type: "script"
+      title: "Deploy kubernetes"
+      script: |-
+        sudo setenforce 0
+        # https://github.com/kubernetes/minikube/blob/master/docs/configuring_kubernetes.md
+        sudo minikube start --vm-driver=none --extra-config=kubelet.cgroup-driver=systemd
+    - type: "script"
+      title: "Build the machine api operator"
+      script: |-
+        # build the mao image
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/machine-api-operator
+        go build -v -o bin/machine-api-operator github.com/openshift/machine-api-operator/cmd
+        go get -u github.com/openshift/imagebuilder/cmd/imagebuilder
+        sudo mv /data/bin/imagebuilder /usr/bin
+        sudo imagebuilder -t "docker.io/machine-api-operator:$(git describe --always --abbrev=7)" .
+    - type: "script"
+      title: "Deploy the machine api operator"
+      script: |-
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/machine-api-operator
+
+        # Drop the tectonic-system ns
+        cat << HEREDOC > namespace.yaml
+        ---
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+          name: tectonic-system
+        HEREDOC
+
+        sudo kubectl apply -f namespace.yaml
+
+        # Drop the CRD definition
+        cat << HEREDOC > appversion-crd.yaml
```

Comment:
Instead of creating these manifests inline, can we store them in the machine-api-operator repo, e.g tests/ so we don't have a dependency on this repo as the operator evolves. Otherwise if the for example the operator config changes, we'll need to update this to get it green

Reply:
I vote for moving this whole script to mao repo.

Reply:
Done